src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -134,7 +135,7 @@
  */
 @SuppressWarnings({"unchecked", "rawtypes"}) // mistakes in various places
 @SuppressFBWarnings("DMI_RANDOM_USED_ONLY_ONCE") // https://github.com/spotbugs/spotbugs/issues/1539
-public abstract class AbstractFolder<I extends TopLevelItem> extends AbstractItem implements TopLevelItem, ItemGroup<I>, ModifiableViewGroup, StaplerFallback, ModelObjectWithChildren, StaplerOverridable {
+public abstract class AbstractFolder<I extends TopLevelItem> extends AbstractItem implements TopLevelItem, ItemGroup<I>, ModifiableViewGroup, StaplerFallback, ModelObjectWithChildren, StaplerOverridable, IconSpec {
```

Comment:
I am sure this works - but this would seem to be incorrect.

`IconSpec` is not changed in the referenced PR and the documentation for it states

> Plugin extension points that implement/extend Action/ManagementLink should also implement this interface.

But this class meets none of those criteria?

Reply:
https://github.com/jenkinsci/jenkins/pull/10049#issuecomment-2537188911

Reply:
resolved by https://github.com/jenkinsci/jenkins/pull/10049/commits/d6868c970a06a17633cf7d145beef28598730a0b

Reply:
I've updated the javadoc, it was already implemented by more than that, I kept it quite general purpose as a lot of things use it